shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java

```diff
@@ -138,4 +207,140 @@ public String getDescription() {
   public void validateArgs(CommandLine cl) throws InvalidArgumentException {
     CommandUtils.checkNumOfArgsNoLessThan(this, cl, 1);
   }
+
+  private class LoadTask implements Callable<Boolean> {
+    private final long mBlockId;
+    private final WorkerNetAddress mDataSource;
+    private final boolean mLocal;
+    private final URIStatus mStatus;
+    private final OpenFilePOptions mOptions;
+
+    LoadTask(long blockId, WorkerNetAddress dataSource, boolean local, URIStatus status,
+        OpenFilePOptions options) {
+      mBlockId = blockId;
+      mDataSource = dataSource;
+      mLocal = local;
+      mStatus = status;
+      mOptions = options;
+    }
+
+    @Override
+    public Boolean call() throws Exception {
+      BlockInfo info = mStatus.getBlockInfo(mBlockId);
+      long blockLength = info.getLength();
+      CacheRequest request = CacheRequest.newBuilder().setBlockId(mBlockId).setLength(blockLength)
+          .setOpenUfsBlockOptions(getOpenUfsBlockOptions(mBlockId, mStatus, mOptions))
+          .setSourceHost(mDataSource.getHost()).setSourcePort(mDataSource.getDataPort()).build();
+      // TODO(jianjian) support FUSE processLocalWorker?
+      WorkerNetAddress worker;
+      try {
+        if (mLocal) {
+          // send request to local worker
+          worker = mFsContext.getNodeLocalWorker();
+        } else { // send request to data source
+          worker = mDataSource;
+        }
+        try (CloseableResource<BlockWorkerClient> blockWorker =
+            mFsContext.acquireBlockWorkerClient(worker)) {
+          CacheResponse response = blockWorker.get().cache(request);
+          return response.getSuccess();
+        }
+      } catch (Exception e) {
+        System.out.printf("Failed to complete cache request for block %d of file %s: %s", mBlockId,
+            mStatus.getPath(), e.toString());
+      }
+      return false;
+    }
+  }
+
+  // todo(jianjian) consider refactoring with AlluxioBlockStore
+  private WorkerNetAddress getDataSouce(long blockId, URIStatus status, BlockLocationPolicy policy)
+      throws IOException {
+    BlockInfo info = status.getBlockInfo(blockId);
+    List<BlockLocation> locations = info.getLocations();
+    List<BlockWorkerInfo> blockWorkerInfo = Collections.EMPTY_LIST;
+    TieredIdentity tieredIdentity =
+        TieredIdentityFactory.localIdentity(mFsContext.getClusterConf());
+    // Initial target workers to read the block given the block locations.
+    Set<WorkerNetAddress> workers;
+    // Note that, it is possible that the blocks have been written as UFS blocks
+    if (status.isPersisted() || status.getPersistenceState().equals("TO_BE_PERSISTED")) {
+      blockWorkerInfo = mFsContext.getCachedWorkers();
+      if (blockWorkerInfo.isEmpty()) {
+        throw new UnavailableException(ExceptionMessage.NO_WORKER_AVAILABLE.getMessage());
```

Comment:
Let's avoid using `ExceptionMessage` . that's a bad practise causing duplicated messages hard to tell where it is from in source code. Just put your own words here

Reply:
Since I reuse the function in AlluxioBlockStore for the loadCommand, do we still want to update the exception message?